### DISCUSSION TITLE
Add the file and frontmatter modules to the tfile prototype?

### DISCUSSION THREAD

**Discussion by DanielJMaxwell** [SilentVoid13/Templater#1489] [Ideas & Feature Requests]
It seems odd that once you access a tfile object you can't then access functions of the file and frontmatter modules that would allow to manipulate it.

For example: if you use 'tp.config.active_file' to get the calling file, you can't then access its frontmatter. You might expect to be able to use a syntax like: 'tp.config.active_file.frontmatter.tags' to fetch details about the file from which the template is called, but currently can't, as far as I can tell.

**Comment by Zachatoo** [comment]
I don't want to add anything to the [TFile](https://docs.obsidian.md/Reference/TypeScript+API/TFile) prototype since that comes straight from Obsidian. I suppose we could add our own abstraction layer, but I frankly don't have time to do something like that, especially given some of the [complexities of updating frontmatter](https://github.com/SilentVoid13/Templater/issues/1387) when functions such as `tp.file.include` are considered.

It is preferred to use Obsidian APIs for interacting with frontmatter, such as [`app.metadataCache.getFileCache`](https://docs.obsidian.md/Reference/TypeScript+API/MetadataCache/getFileCache) and [`app.fileManager.processFrontMatter`](https://docs.obsidian.md/Reference/TypeScript+API/FileManager/processFrontMatter).
